README.md
# SWWF (Setup Windows without f@#k) 

Script that setup Windows 10 to my personal prefrences. 

## Works on: 
- Windows 10 LTSC 
- Windows 10 Pro 
- Windows 11 Pro

## Installation

Just put all files from archive to C drive and run as administrator

## Launch

**Recommended:** Run PostInstall.cmd as an Administrator

If you want to set settings in PersonalSetup.ps1 launch with "y" argument. It is installs some preferences for my portable apps, so you **do not need** to use it.
## Links

Main setup took from Izzy Laif "Настраиваю Windows 10 часов". Link: https://www.youtube.com/watch?v=ATJbkPDLOVE&t=3s

Setup for perfomance and scripts took from: Windows 10 & 11 TURBO Optimization, Fr33thy. Link: https://www.youtube.com/watch?v=Q1zLtq491ZE&t=673s


Changelog.md
# v1.2.1

## Changes
1. Disable Network Discovery and Sharing
2. Update Microsoft Edge path to remove

## Fixes
1. Fix removing arrow from shortcuts


# 1.2.0

## Changes

1. Update registry in personal setup
2. Now script doesn't ask for PersonalSetup.ps1 launch. To activate it, you need to launch script with 'y' args.
3. Remove AMD Setting Optimization

## Fixes

1. Increase time for Debloat.ps1.

# v1.1.1

## Changes

1. Add more information about progress of each step
2. Add step that disable Windows Defeder
3. Add administration rights check
4. Files deletes in the end of script

## Fixes

1. Remove some mistakes in PersonalSetup.ps1

# v1.1.0

## Changes

1. Update to latest fr33thy Turbo Optimization. Edit it to my preferences
2. Add .NET installation file

## Fixes

1. FiraCode NFM font installing correctly

# v1.0.1

## Changes

1. Separate personal and common settings
2. Sort to category registry tweaks
3. Combine scripts into one.

# v1.0.0

## Changes

1. \_AppSetup. Add WinRAR activation
2. \_AppSetup. Return Firefox Portable to default browser
3. \_DebloatScript. Remove all Xbox stuff
4. \_Registry. Add Capslock to change layout
5. \_Registry. Remove all shortcut from dekstop

## Fixes

1. \_PostInstall. Now fonts and keyboard setup correctly

# v0.9.2

## Changes

1. Fix bugs

# v0.9.1

## Changes

1. Remove icon from taskbar. Now I use toolbar
2. Add path to nvim

# v0.9

## Changes

1. Remove unused apps from \_AppSetup.ps1
2. Change keyboard layout now at CAPS

# v0.8.3

## Changes

1. \_PostInstall. Now check that user run script as administrator
2. Add Telegram.reg

# v0.8.2

## Changes

1. \_AppSetup. All reg files now run silent

## Fixes

1. \_PostInstall. Word.exe now restore from Windows Defender

# v0.8.1

## Changes

1. \_AppSetup. Enable preview pane

## Fixes

1. \_PostInstall. Word.exe now restore from Windows Defender

# v0.8

## Changes

1. \_AppSetup. Check that K drive is mounted, if not launch truecrypt.bat and wait
2. \_AppSetup. OfficePortable add to Windows Defender exclusion
3. \_AppSetup. Add reg file to restore WinRAR settings
4. \_AppSetup. Add Telegram.reg files. Remove from \_Registry.reg
5. \_AppSetup. Add Studio One and Melodyne setup
6. \_AppSetup. Git setups automatically
7. \_Registry.reg. Timezone UTC+2 Kyiv

# v0.7.1

## Changes

1. Add registry that disable widgets in Windows 11. Free at least 1.5 Gb of RAM
2. \_DebloatScript. Add XboxApp and Teams to uninstall

## Fixes

1. Now \_PostInstall.cmd can run from any path
2. Enviromental variables adds correctly
3. Now coping shortcut avaliable for every user
4. Block Uninstall bloat and Setup my apps now proceed correctly
5. Now Microsoft Edge uninstalling
6. Execution police for powershell now applied in cmd file
7. \_AppSetup and \_DeblaotScript now launch
8. Now script wait until C++ and DX to be installed

# v0.7

## Changes

1. Move to PowerShell
2. Add **_\_DebloatScript.ps1_** to uninstall blotware
3. Add **_\_AppSetup.ps1_** to setup my apps
4. Move enviroment variable to script for better customization

# v0.6

## Changes

1. Uninstall Microsoft Edge in Windows 11
2. Remove disabling Windows Defender in bat and reg files. Try to find better way
3. Remove disable auto update. This block settings in Windows Update


_AppSetup.ps1
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------TEST K DRIVE--------------------------------------
Write-Host ---------------------------------------------------------------------------------
$check_k_drive = Test-Path -Path "K:\"
if ($check_k_drive -eq $false) {
    C:\BATCH\truecrypt.bat
    Wait-Process VeraCrypt-x64
}
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -----------------------------RESTORE WINRAR SETTINGS-----------------------------
Write-Host ---------------------------------------------------------------------------------
regedit.exe /s "C:\Windows\_WinRAR.reg"
New-Item -Path "C:\Users\Administrator\AppData\Roaming" -Name "WinRAR" -ItemType "directory" 
Copy-Item -Path "C:\Windows\rarreg.key" -Destination "C:\Users\Administrator\AppData\Roaming\WinRAR"
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ----------------------------MAKE FIREFOX PORTABLE DEFAULT------------------------
Write-Host ---------------------------------------------------------------------------------
Set-Location K:\FirefoxPortable
.\FirefoxPortable2DefaultPrograms.bat
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -----------------------------ADD TELEGRAM EXTENSION------------------------------
Write-Host ---------------------------------------------------------------------------------
regedit.exe /s "C:\Windows\_Telegram.reg"
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------SETUP GIT-----------------------------------------
Write-Host ---------------------------------------------------------------------------------
C:\PortableApps\Git\bin\git.exe config --global user.name "Pixaler"
C:\PortableApps\Git\bin\git.exe config --global user.email "gnidko02@gmail.com"
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Exit

Files/Debloat.ps1
Get-AppxPackage -allusers Clipchamp.Clipchamp | Remove-AppxPackage
Get-AppxPackage -allusers Disney.37853FC22B2CE | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.549981C3F5F10 | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.BingNews | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.BingWeather | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.GetHelp | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Getstarted | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MSPaint | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Microsoft3DViewer | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MicrosoftOfficeHub | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MicrosoftSolitaireCollection | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MicrosoftStickyNotes | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MixedReality.Portal | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Office.OneNote | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.OneDriveSync | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.People | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.PowerAutomateDesktop | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.ScreenSketch | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.SkypeApp | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Todos | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Wallet | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsAlarms | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsCalculator | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsCamera | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsFeedbackHub | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsMaps | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsSoundRecorder | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.YourPhone | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.ZuneMusic | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.ZuneVideo | Remove-AppxPackage
Get-AppxPackage -allusers MicrosoftCorporationII.QuickAssist | Remove-AppxPackage
Get-AppxPackage -allusers MicrosoftTeams | Remove-AppxPackage
Get-AppxPackage -allusers MicrosoftWindows.Client.WebExperience | Remove-AppxPackage
Get-AppxPackage -allusers SpotifyAB.SpotifyMusic | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsCommunicationsApps | Remove-AppxPackage
C:\Windows\SysWOW64\OneDriveSetup.exe -uninstall
C:\Users\Administrator\AppData\Local\Microsoft\OneDrive\22.012.0117.0003\OneDriveSetup.exe /uninstall
Get-WindowsPackage -Online | Where PackageName -like *Hello-Face* | Remove-WindowsPackage -Online -NoRestart
Get-WindowsPackage -Online | Where PackageName -like *QuickAssist* | Remove-WindowsPackage -Online -NoRestart
Get-ProvisionedAppxPackage -Online | `
Where-Object { $_.PackageName -match "xbox" } | `
ForEach-Object { Remove-ProvisionedAppxPackage -Online -AllUsers -PackageName $_.PackageName }
C:\"Program Files (x86)"\Microsoft\Edge\Application\9*\Installer\setup.exe --uninstall --force-uninstall --system-level
C:\"Program Files (x86)"\Microsoft\Edge\Application\10*\Installer\setup.exe --uninstall --force-uninstall --system-level


_DebloatScript.ps1
Get-AppxPackage -allusers Disney.37853FC22B2CE | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.549981C3F5F10 | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.BingNews | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.BingWeather | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.GetHelp | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Getstarted | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Microsoft3DViewer | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MicrosoftOfficeHub | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MicrosoftSolitaireCollection | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MicrosoftStickyNotes | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MixedReality.Portal | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MSPaint | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Office.OneNote | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.OneDriveSync | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.People | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.PowerAutomateDesktop | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.SkypeApp | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Todos | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Wallet | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsAlarms | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsCamera | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsFeedbackHub | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsMaps | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsSoundRecorder | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.YourPhone | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.ZuneMusic | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.ZuneVideo | Remove-AppxPackage
Get-AppxPackage -allusers MicrosoftTeams | Remove-AppxPackage
Get-AppxPackage -allusers MicrosoftWindows.Client.WebExperience | Remove-AppxPackage
Get-AppxPackage -allusers SpotifyAB.SpotifyMusic | Remove-AppxPackage
Get-AppxPackage -allusers microsoft.windowscommunicationsapps | Remove-AppxPackage
Get-AppxPackage -allusers MicrosoftTeams* | Remove-AppxPackage
# Remove all xbox stuff
Get-ProvisionedAppxPackage -Online | `
Where-Object { $_.PackageName -match "xbox" } | `
ForEach-Object { Remove-ProvisionedAppxPackage -Online -AllUsers -PackageName $_.PackageName }
C:\Windows\SysWOW64\OneDriveSetup.exe -uninstall
Set-Location -Path "C:\Program Files (x86)\Microsoft\Edge\Application\9*\Installer" 
./setup.exe --uninstall --force-uninstall --system-level
Get-WindowsPackage -Online | Where PackageName -like *Hello-Face* | Remove-WindowsPackage -Online -NoRestart
Get-WindowsPackage -Online | Where PackageName -like *QuickAssist* | Remove-WindowsPackage -Online -NoRestart
Exit

_PersonalSetup.ps1
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------TEST K DRIVE--------------------------------------
Write-Host ---------------------------------------------------------------------------------
$check_k_drive = Test-Path -Path "K:\"
if ($check_k_drive -eq $false) {
    C:\BATCH\truecrypt.bat
    Wait-Process VeraCrypt-x64
}
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------CREATE FOLDERS------------------------------------
Write-Host ---------------------------------------------------------------------------------
New-Item -Path "C:\" -Name "TEMP" -ItemType "directory" 
New-Item -Path "C:\" -Name "!DOWNLOADS" -ItemType "directory" 
New-Item -Path "C:\" -Name "!APPS" -ItemType "directory" 
New-Item -Path "C:\" -Name "!GAMES" -ItemType "directory" 
New-Item -Path "C:\" -Name "!MYCOR" -ItemType "directory" 
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------INSTALL FONTS-------------------------------------
Write-Host ---------------------------------------------------------------------------------
Copy-Item -Path "C:\Windows\hack.ttf" -Destination "C:\Windows\Fonts"
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host --------------------------EDIT ENVIROMENTAL VARIABLES----------------------------
Write-Host ---------------------------------------------------------------------------------
[Environment]::SetEnvironmentVariable("Path", $env:PATH + ";C:\BATCH;C:\PortableApps\VSCode\bin;C:\PortableApps\Git\bin;C:\PortableApps\nvim\bin", [System.EnvironmentVariableTarget]::User)
[Environment]::SetEnvironmentVariable("Path", $env:PATH + ";C:\BATCH;C:\PortableApps\VSCode\bin;C:\PortableApps\Git\bin;C:\PortableApps\nvim\bin",  [System.EnvironmentVariableTarget]::Machine)
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------TOOLBAR LINKS-------------------------------------
Write-Host ---------------------------------------------------------------------------------
Copy-Item -Path "C:\BATCH\Links\*" -Destination "C:\Users\$env:USERNAME\Favorites\Links" -Recurse
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ----------------------------IMPORTING PERSONAL REGISTRY TWEAKS----------------------------
Write-Host ---------------------------------------------------------------------------------
regedit.exe /s "C:\Windows\_PersonalRegistry.reg"
Stop-Process -name explorer -force
Write-Host .
Write-Host .
Start-Sleep -Seconds 15
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------DISABLE SERVICES----------------------------------
Write-Host ---------------------------------------------------------------------------------
Stop-Service -Name "ShellHWDetection" -Force
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ----------------------------CAPS LIKE CHANGE LAYOUT------------------------------
Write-Host ---------------------------------------------------------------------------------
C:\Windows\engcaps\setup.exe 
Start-Sleep -Seconds 5
Stop-Process -name setup -force
C:\Windows\ruscaps\setup.exe 
Start-Sleep -Seconds 5
Stop-Process -name setup -force
C:\Windows\ukrcaps\setup.exe 
Start-Sleep -Seconds 5
Stop-Process -name setup -force
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -----------------------------RESTORE WINRAR SETTINGS-----------------------------
Write-Host ---------------------------------------------------------------------------------
regedit.exe /s "C:\Windows\_WinRAR.reg"
New-Item -Path "C:\Users\Administrator\AppData\Roaming" -Name "WinRAR" -ItemType "directory" 
Copy-Item -Path "C:\Windows\rarreg.key" -Destination "C:\Users\Administrator\AppData\Roaming\WinRAR"
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ----------------------------MAKE FIREFOX PORTABLE DEFAULT------------------------
Write-Host ---------------------------------------------------------------------------------
Set-Location K:\FirefoxPortable
.\FirefoxPortable2DefaultPrograms.bat
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -----------------------------ADD TELEGRAM EXTENSION------------------------------
Write-Host ---------------------------------------------------------------------------------
regedit.exe /s "C:\Windows\_Telegram.reg"
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------SETUP GIT-----------------------------------------
Write-Host ---------------------------------------------------------------------------------
C:\PortableApps\Git\bin\git.exe config --global user.name "Pixaler"
C:\PortableApps\Git\bin\git.exe config --global user.email "gnidko02@gmail.com"
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Exit


Files/PostInstall.ps1
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------EDITING SWAP--------------------------------------
Write-Host ---------------------------------------------------------------------------------
# Uncheck automatically set pagefile
$pagefile = Get-WmiObject Win32_ComputerSystem -EnableAllPrivileges
$pagefile.AutomaticManagedPagefile = $false
$pagefile.put() | Out-Null
# Set 8000 Mb pagefile on C drive
$pagefileset = Get-WmiObject Win32_pagefilesetting
$pagefileset.InitialSize = 8000
$pagefileset.MaximumSize = 8000
$pagefileset.Put() | Out-Null
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ----------------------------REMOVE ARROW FROM SHORTCUT----------------------
Write-Host ---------------------------------------------------------------------------------
Copy-Item -Path "C:\Files\Packages\Blank.ico" -Destination "C:\Windows"
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ----------------------------DISABLING NETWORK DISCOVERY--------------------------
Write-Host ---------------------------------------------------------------------------------
netsh advfirewall firewall set rule group="Network Discovery" new enable=No
netsh advfirewall firewall set rule group="File and Printer Sharing" new enable=No
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
$setup_my_apps = $args[0]
if ($setup_my_apps -like 'y') {
    C:\Files\PersonalSetup.ps1
}
else {
    Start-Sleep -Seconds 1
}
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .


_PostInstall.ps1
Start-Sleep -Seconds 2
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------YOU RUNNING AS ADMIN?-----------------------------
Write-Host ---------------------------------------------------------------------------------
$currentPrincipal = New-Object Security.Principal.WindowsPrincipal([Security.Principal.WindowsIdentity]::GetCurrent())
if ($currentPrincipal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator) -eq $false)
{
    Write-Host .
    Write-Host .
    Write-Host Please run as administrator!
    Start-Sleep -Seconds 2
    Write-Host .
    Write-Host .
    Exit
}
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------EDITING SWAP--------------------------------------
Write-Host ---------------------------------------------------------------------------------
# Uncheck automatically set pagefile
$pagefile = Get-WmiObject Win32_ComputerSystem -EnableAllPrivileges
$pagefile.AutomaticManagedPagefile = $false
$pagefile.put() | Out-Null
# Set 8000 Mb pagefile on C drive
$pagefileset = Get-WmiObject Win32_pagefilesetting
$pagefileset.InitialSize = 8000
$pagefileset.MaximumSize = 8000
$pagefileset.Put() | Out-Null
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -----------------------------INSTALLING POWER PLAN-------------------------------
Write-Host ---------------------------------------------------------------------------------
powercfg -import "C:\Windows\_BitsumHighestPerformance.pow" 77777777-7777-7777-7777-777777777777
powercfg -SETACTIVE "77777777-7777-7777-7777-777777777777"
powercfg -delete 381b4222-f694-41f0-9685-ff5bb260df2e
powercfg -delete 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c 
powercfg -delete a1841308-3541-4fab-bc81-f71556f20b4a
powercfg -attributes SUB_PROCESSOR 75b0ae3f-bce0-45a7-8c89-c9611c25e100 -ATTRIB_HIDE
powercfg -setdcvalueindex SCHEME_CURRENT 54533251-82be-4824-96c1-47b60b740d00 75b0ae3f-bce0-45a7-8c89-c9611c25e100 3000
powercfg -setacvalueindex SCHEME_CURRENT 54533251-82be-4824-96c1-47b60b740d00 75b0ae3f-bce0-45a7-8c89-c9611c25e100 3000
powercfg -change -monitor-timeout-ac 20
powercfg -change -monitor-timeout-dc 20
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------DISABLING HIBERNATE-------------------------------
Write-Host ---------------------------------------------------------------------------------
powercfg -h off
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------UNINSTALL BLOAT-----------------------------------
Write-Host ---------------------------------------------------------------------------------
Write-Host .
Write-Host .
Get-AppxPackage -allusers Disney.37853FC22B2CE | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.549981C3F5F10 | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.BingNews | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.BingWeather | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.GetHelp | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Getstarted | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Microsoft3DViewer | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MicrosoftOfficeHub | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MicrosoftSolitaireCollection | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MicrosoftStickyNotes | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MixedReality.Portal | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.MSPaint | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Office.OneNote | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.OneDriveSync | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.People | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.PowerAutomateDesktop | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.SkypeApp | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Todos | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.Wallet | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsAlarms | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsCamera | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsFeedbackHub | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsMaps | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.WindowsSoundRecorder | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.YourPhone | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.ZuneMusic | Remove-AppxPackage
Get-AppxPackage -allusers Microsoft.ZuneVideo | Remove-AppxPackage
Get-AppxPackage -allusers MicrosoftTeams | Remove-AppxPackage
Get-AppxPackage -allusers MicrosoftWindows.Client.WebExperience | Remove-AppxPackage
Get-AppxPackage -allusers SpotifyAB.SpotifyMusic | Remove-AppxPackage
Get-AppxPackage -allusers microsoft.windowscommunicationsapps | Remove-AppxPackage
Get-AppxPackage -allusers MicrosoftTeams* | Remove-AppxPackage
# Remove all xbox stuff
Get-ProvisionedAppxPackage -Online | `
Where-Object { $_.PackageName -match "xbox" } | `
ForEach-Object { Remove-ProvisionedAppxPackage -Online -AllUsers -PackageName $_.PackageName }
C:\Windows\SysWOW64\OneDriveSetup.exe -uninstall
Set-Location -Path "C:\Program Files (x86)\Microsoft\Edge\Application\9*\Installer" 
./setup.exe --uninstall --force-uninstall --system-level
Get-WindowsPackage -Online | Where PackageName -like *Hello-Face* | Remove-WindowsPackage -Online -NoRestart
Get-WindowsPackage -Online | Where PackageName -like *QuickAssist* | Remove-WindowsPackage -Online -NoRestart
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------SETUP MY SETTINGS-------------------------------------
Write-Host ---------------------------------------------------------------------------------
Write-Host "Setup my setting? (y/n)"
$setup_my_apps = Read-Host -Prompt "You options is"
if ($setup_my_apps -like 'y') {
    C:\_PersonalSetup.ps1
}
if ($setup_my_apps -like 'n') {
    Start-Sleep -Seconds 1
}
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ----------------------------------INSTALLING C++---------------------------------
Write-Host ---------------------------------------------------------------------------------
C:\Windows\_C++.exe /S
Wait-Process _C++
Write-Host .
Write-Host .
Write-Host .
Write-Host .
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -----------------------------------INSTALLING DX---------------------------------
Write-Host ---------------------------------------------------------------------------------
C:\Windows\_DX\DXSETUP.exe /silent
Wait-Process DXSETUP
Write-Host .
Write-Host .
Start-Sleep -Seconds 10
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ----------------------------IMPORTING REGISTRY TWEAKS----------------------------
Write-Host ---------------------------------------------------------------------------------
regedit.exe /s "C:\Windows\_Registry.reg"
Stop-Process -name explorer -force
Write-Host .
Write-Host .
Start-Sleep -Seconds 15
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ----------------------------DISABLING NETWORK DISCOVERY--------------------------
Write-Host ---------------------------------------------------------------------------------
netsh advfirewall firewall set rule group="Network Discovery" new enable=No
netsh advfirewall firewall set rule group="File and Printer Sharing" new enable=No
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host -------------------------------CLEAR TEMP FOLDERS--------------------------------
Write-Host ---------------------------------------------------------------------------------
Get-ChildItem -Path "C:\Windows\Temp" *.* -Recurse | Remove-Item -Force -Recurse
Get-ChildItem -Path "$env:TEMP" *.* -Recurse | Remove-Item -Force -Recurse
Get-ChildItem -Path "C:\Users\$env:UserName\AppData\Local\Temp" *.* -Recurse | Remove-Item -Force -Recurse
Write-Host .
Write-Host .
Start-Sleep -Seconds 2
Write-Host .
Write-Host .
Write-Host ---------------------------------------------------------------------------------
Write-Host ------------------------------------RESTARTING PC--------------------------------
Write-Host ---------------------------------------------------------------------------------
Write-Host .
Write-Host .
Start-Sleep -Seconds 5
Remove-Item C:\Windows\_DX -Recurse
Remove-Item C:\Windows\engcaps -Recurse
Remove-Item C:\Windows\ruscaps -Recurse
Remove-Item C:\Windows\ukrcaps -Recurse
Remove-Item C:\Windows\_BitsumHighestPerformance.pow
Remove-Item C:\Windows\_C++.exe
Remove-Item C:\Windows\_Registry.reg
Remove-Item C:\Windows\_PersonalRegistry.reg
Remove-item C:\Windows\hack.ttf
Remove-item C:\Windows\rarreg.key
Remove-Item C:\_PostInstall.cmd
Remove-Item C:\_PersonalSetup.ps1
Restart-Computer
Remove-Item C:\_PostInstall.ps1
